### DISCUSSION TITLE
Typescript support?

### DISCUSSION THREAD

**Discussion by hally9k** [TanStack/query#231] [General] [answered]
At first glance I don't see a tsconf here, is there a typescript support story for this?

**Comment by tannerlinsley** [comment]
React Query is not written in TS, but there are community types in Definitely Typed that you can reference.

**Comment by mskelton** [comment]
There is an active PR for updating `@types/react-query` with the changes since the release of react-query v1.  Once that is merged, you should be able to use v1 with TypeScript very easily.

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/42830

Until that is merged/released, you can add the following code to a `react-query.d.ts` file with the `...` being replaced by the type definitions in the PR I just mentioned.

```ts
declare module 'react-query' {
   ...
}
```

**Comment by hally9k** [comment, marked as answer]
@mskelton I will check out the community types thanks for the pointers 😄

**Comment by tannerlinsley** [comment]
FYI https://github.com/tannerlinsley/react-query/discussions/362
